### PR TITLE
Remove dropout in embedding layer of OPT

### LINKED ISSUE #18844: Dropout in OPT embedding layer
### System Info

main

### Who can help?

@ArthurZucker, @patrickvonplaten, @LysandreJik

### Information

- [X] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

The [OPT paper](https://arxiv.org/pdf/2205.01068.pdf) (sec 2.2) mentioned 
> We use a dropout of 0.1 throughout, but we do not apply any dropout to embeddings.

This is also supported by loading [the official checkpoints](https://github.com/facebookresearch/metaseq/tree/main/projects/OPT) and running the following check

```python
import torch
data = torch.load("reshard-model_part-0.pt")
assert data['cfg']['model'].no_emb_dropout
```

However, in `modeling_opt.py`, dropout is applied to the embedding layer.

https://github.com/huggingface/transformers/blob/80367cd1fb6d36ca6bdd99b70586aab4ffae1ae1/src/transformers/models/opt/modeling_opt.py#L641-L642

### Expected behavior

No dropout in the embedding layer

### PR DESCRIPTION
**Pull request by shijie-wu**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/issues/18844

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by LysandreJik** [comment]
Thanks for your PR @shijie-wu! Pinging @ArthurZucker and @younesbelkada to take a look at this as soon as they're back from leave (in about a week's time).

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks for the catch. Might have to warn users who are currently training a model and rely on this dropout (it is not breaking but might have an impact)

**Review by younesbelkada — APPROVED** [review]
LGTM thanks for the nice catch!
I don't see any major breaking changes here. The only possible breaking change case would be for someone who is training an OPT model with the previous version (ie with the dropout on the embedding layer) and should not be able to reproduce the same training dynamics with this proposed fix. But at inference and testing time dropout should not be used so this should have no impact in this matter
